qemu/tests/nic_hotplug.py

```diff
@@ -99,17 +119,31 @@ def run(test, params, env):
 
             error.context("Ping guest's new ip from host", logging.info)
             if params.get("additional_operation"):
-                s_session.cmd_output_safe("route add %s dev %s" %
-                                          (host_ip_addr, ifname))
+                add_route_cmd = "route add %s dev %s" % (host_ip_addr, ifname)
+                s_session.cmd_output_safe(add_route_cmd)
+
             status, output = utils_test.ping(hotnic_ip, 10, timeout=30)
             if status:
                 err_msg = "New nic failed ping test, error info: '%s'"
                 raise error.TestFail(err_msg % output)
 
-            error.context("Ping guest's new ip after pasue/resume",
-                          logging.info)
+            error.context("Reboot vm after hotplug nic", logging.info)
+            # reboot vm via serial port since some guest can't auto up
+            # hotplug nic and next step will check is hotplug nic works.
+            s_session = vm.reboot(session=s_session, serial=True)
+            vm.verify_alive()
+            if guest_is_linux:
+                s_session.cmd("dhclient -r", timeout=120)
+                s_session.cmd("dhclient %s" % ifname, timeout=240)
+            else:
+                s_session.cmd("ipconfig /renew all", timeout=240)
+            hotnic_ip = vm.wait_for_get_address(nic_name)
```

Comment:
Hi,@xutian
1.suggest add context with  "error.context("Check if new interface gets ip address after reboot", logging.info)"
2.wait_for_get_address(nic_name) should add timeout.
3.if failed to get guest's address,please provide error message such as "err_msg = "Could not get or verify nic ip address""
4.if failed to get guest's address ,should add check code such as line 95-116.

thanks,
xiagao